rust/src/bin/lq.rs

```diff
@@ -168,18 +168,22 @@ async fn create_index(
     metric_type: &Option<String>,
     use_opq: bool,
 ) -> Result<()> {
-    let col = column
-        .as_ref()
-        .ok_or_else(|| Error::Index("Must specify column".to_string()))?;
-    let _ = index_type.ok_or_else(|| Error::Index("Must specify index type".to_string()))?;
+    let col = column.as_ref().ok_or_else(|| Error::Index {
```

Comment:
This does make more typing than before, can we add a few methods, such as 

```
impl Error {
   fn index(msg: &str) -> Self {}
}
```
and etc.  Otherwise need to check out the internal structure of each Error type everytime.